tools/haskell/README.md

```diff
@@ -10,3 +10,35 @@ from `cabal` will interfere with the protocol.
 
 [`ad`]: https://hackage.haskell.org/package/ad
 [haskell]: https://haskell.org/
+
+## Running outside of Docker
+
+You need a Haskell development environment, including `cabal` and
+`ghc`. The easiest way to get one is to use [ghcup][] or
+[shell.nix][].
+
+Then run the following command to populate local knowledge of the
+Hackage package repository:
+
+```
+$ cabal update
```

Comment:
I get an error when I run this in the Nix config:

```
zsh: command not found: cabal
```

Reply:
Oh yeah, I always have this stuff in my global Nix config. Fixed.